### ISSUE TITLE
Ignores `nocase` when searching for non-glob

### ISSUE THREAD

**Issue by phated** [js-cli/node-findup-sync#34]
The problem lies in using `exists` in https://github.com/js-cli/node-findup-sync/blob/master/index.js#L74-L91 but I'm not sure how to deal with that.

**Comment by phated** [comment]
Opened for tracking  /cc @jonschlinkert

**Comment by jonschlinkert** [comment]
noted, we're on a linux box now trying to figure out what's happening

**Comment by jonschlinkert** [comment]
Okay, [this code in liftoff](https://github.com/js-cli/js-liftoff/blob/master/lib/find_config.js#L21) and similar code in `exists` will need to change. 

The most reliable solution I can think of is to call `fs.readdirSync` in each directory then loop over the files and lowercase both the given pattern (non-glob) and each file path. 

Obviously, this solution would have the opposite disadvantage of _not_ returning exact matches, and it would be orders of magnitude slower, since we would be going from checking for a specific file to iterating over 1 to n files, but if it's necessary I can't see a way way around it.

Here are some solutions to mitigate the performance hit:

- check for platform, and only do this on linux 
- add a `nocase` option, giving users control over whether or not to iterate over all files in every directory
- check for the file path, exactly as given. if not found, check for lowercase, then as a last resort loop over the files in the directory.

IMHO looping over files in each directory should be the last resort. We could probably check every uppercase and lowercase permutation on a given pattern and it would still be faster than looping over directories.

I'm going to start working on this. in the meantime any thoughts/feedback/preferences would be appreciated.

**Comment by phated** [comment]
I think the original implementation crawled the filesystem. Why did we split the logic between globs and non-globs?

**Comment by jonschlinkert** [comment]
sorry @doowb and I spent a while going through the code, trying scenarios. I'll be working on this more tomorrow until it's fixed.

> Why did we split the logic between globs and non-globs?

fwiw this issue seems to effect glob as well (at least it seems like the same issue?):

- https://github.com/isaacs/node-glob/issues/161
- https://github.com/isaacs/node-glob/issues/237
- https://github.com/isaacs/node-glob/issues/238

to answer your question:

for a couple of reasons, but mainly because we can determine with a high degree of certainty when a glob is passed versus a non-glob and, given that, we should usually be able to do anything glob would do with a filepath, but faster. Sometimes exponentially faster - the speed of `readdir` is effected by the number of files in a directory, whereas checking for a single file isn't. e.g. why read an entire directory when we already know which file we need. The other reason is that "finding up" is a bit different than regular glob searching, because we search up, and recurse down (I [explain more here](https://github.com/js-cli/node-findup-sync/issues/31#issuecomment-208611929)). So it gets slower faster.

All of that said, I'm not attached to any code, if other needs outweigh the choices made, we can make whatever changes are necessary. 

to clarify though, the bug is both in liftoff and `fs-exists-sync`. even if globbing had been implemented everywhere in findup-sync, I believe the issue would still exist because of the code in liftoff.
